### PR TITLE
Add credential deleting to the profile

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- load the new profile and navigate to the credential list
- click Delete in the context menu for a credential
- click confirm
- the credential should get removed

**Comment by huwshimi** [comment]
[Add credential deleting to the profile](https://trello.com/c/RQunIXXj/291-add-credential-deleting-to-the-profile)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8481/

File: jujugui/static/gui/src/app/components/profile/credential-list/add/add.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+
+const Popup = require('../../../popup/popup');
+
+class ProfileCredentialListAdd extends React.Component {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/profile/credential-list/add/add.js]: Was this supposed to be delete?

Comment (huwshimi) [reply jujugui/static/gui/src/app/components/profile/credential-list/add/add.js]: Haha, yes, what happened to me?

File: jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js [hunk 2]
```diff
@@ -199,6 +209,16 @@ class ProfileCredentialList extends React.Component {
     this._toggleAdd();
   }
 
+  /**
+    Handle a credential having been created.
+    @param credential {String} The name of the newly created credential.
+  */
+  _onCredentialDeleted(credential) {
+    // Remove the credential from the list.
+    this.state.credentialMap.delete(credential);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js]: Do we want to do this? I think I'd prefer we refresh the list.

File: jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js [hunk 3]
```diff
@@ -220,7 +240,7 @@ class ProfileCredentialList extends React.Component {
         controllerIsReady={this.props.controllerIsReady}
         credential={overrides.credential}
         credentials={
-          credentials ? Array.from(credentials).map(credential => credential.name) : []}
+          credentials ? Array.from(credentials).map(credential => credential[0]) : []}
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js]: doesn't this return the tag and not the display name?

File: jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js [hunk 4]
```diff
@@ -229,6 +249,23 @@ class ProfileCredentialList extends React.Component {
...
+    if (!credential) {
+      return null;
+    }
+    return (
+      <ProfileCredentialListDelete
+        addNotification={this.props.addNotification}
+        credential={credential}
+        onCancel={this._setDeleteCredential.bind(this)}
+        onCredentialDeleted={this._onCredentialDeleted.bind(this)}
+        revokeCloudCredential={this.props.controllerAPI .revokeCloudCredential} />);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js]: space before .

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8487/

File: jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js (hunk 3 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js]: Yep! That's what we want.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8488/

**Comment by huwshimi** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**